package/piksi_system_daemon/src/main.c

```diff
@@ -580,6 +605,14 @@ int main(void)
 
   settings_type_t settings_type_ip_mode;
   pk_settings_register_enum(settings_ctx, ip_mode_enum_names, &settings_type_ip_mode);
+  pk_settings_register(settings_ctx,
```

Comment:
This should be registered last to avoid the interface "bouncing" during boot-up.  Settings values are populated in the order that they are registered, so in order to avoid re-configuring the interface 4 times (once for each config field) we can simply register this field last.

Reply:
There is a flag `eth_settings_initialized` set to true at the end of the settings registration; initially false. While false, the interface is not reconfigured.

This way the `mode` settings appears as the first option.